src/firefly/java/edu/caltech/ipac/firefly/server/query/lc/PtfFileGroupsProcessor.java

```diff
@@ -0,0 +1,204 @@
+                // look for in_ra and in_dec returned by IBE
+                String subLon = dgData.get(rowIdx, "in_ra") != null ? String.format("%.4f", (Double) dgData.get(rowIdx, "in_ra")) : null;
+                if (StringUtils.isEmpty(subLon)) {
+                    subLon = String.format("%.4f", (Double) dgData.get(rowIdx, "ra"));
+                }
+
+                // look for in_ra and in_dec returned by IBE
+                String subLat = dgData.get(rowIdx, "in_dec") != null ? String.format("%.4f", (Double) dgData.get(rowIdx, "in_dec")) : null;
+                if (StringUtils.isEmpty(subLat)) {
+                    // if it fails, try using crval2
+                    subLat = String.format("%.4f", (Double) dgData.get(rowIdx, "dec"));
+                }
+                String cutoutInfo = "_ra" + subLon + "_dec" + subLat + "_asec" + sizeAsecStr;
+
+                extName = extName.replace(".fits",cutoutInfo+".fits");
+                String url = createCutoutURLString_l2(baseUrl, pfilename, subLon, subLat, subSize);
+                logger.briefInfo("cutout url: " + url);
+
+
+//                // strip out filename when using file resolver
+//                if (doFolders) {
+//                    int idx = extName.lastIndexOf("/");
+//                    idx = idx < 0 ? 0 : idx;
+//                    extName = extName.substring(0, idx) + "/";
+//                } else {
+//                    extName = null;
+//                }
+
+                fi = new FileInfo(url, extName, 100000);
+            } else {
+                String url = baseUrl + fileName;
+                fi = new FileInfo(url, extName, 100000);
+            }
+            if (fi != null) {
+                fi.setCookies(cookies);
+                fiArr.add(fi);
+                fgSize += fi.getSizeInBytes();
+            }
+        }
+
+        FileGroup fg = new FileGroup(fiArr, null, fgSize, "PTF Download Files");
+        ArrayList<FileGroup> fgArr = new ArrayList<FileGroup>();
+        fgArr.add(fg);
+        return fgArr;
+    }
+
+    static String createCutoutURLString_l2(String baseUrl, String baseFile, String lon, String lat, String size) {
+
+        //http://irsa.ipac.caltech.edu/ibe/data/ptf/images/level1/{$pfilename}?center=${ra},${dec}&size=${cutoutSizeInDeg}&gzip=false
+
+        String url = baseUrl + baseFile;
+        url += "?center=" + lon + "," + lat;
+        if (!StringUtils.isEmpty(size)) {
+            url += "&size=" + size;
+        }
+        url += "&gzip=" + false;
+
+        return url;
+    }
+
+//    static String getBaseURL(ServerRequest sr) {
+//
+//        String host = sr.getSafeParam("host") != null ? sr.getSafeParam("host") : PtfIbeResolver.PTF_IBE_HOST;
+//
+//        return QueryUtil.makeUrlBase(host) + "/data/ptf/images/level1/";
+//    }
+
+    private static String getBaseURL(ServerRequest sr) throws MalformedURLException {
+        // build service
+        return new PtfFileRetrieve().getBaseURL(sr);
```

Comment:
left from previous code introduced in different access, can be static now. Will change.